packages/bot-web-ui/src/components/bot-notification-messages/__tests__/bot-notification-messages.spec.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { mockStore, StoreProvider } from '@deriv/stores';
+import { render, screen } from '@testing-library/react';
+import { mock_ws } from 'Utils/mock';
+import RootStore from 'Stores/root-store';
+import { DBotStoreProvider, mockDBotStore } from 'Stores/useDBotStore';
+import BotNotificationMessages from '../bot-notification-messages';
+
+jest.mock('@deriv/bot-skeleton/src/scratch/blockly', () => jest.fn());
+jest.mock('@deriv/bot-skeleton/src/scratch/dbot', () => ({
+    saveRecentWorkspace: jest.fn(),
+    unHighlightAllBlocks: jest.fn(),
+}));
+jest.mock('@deriv/bot-skeleton/src/scratch/hooks/block_svg', () => jest.fn());
+
+describe('BotNotificationMessages', () => {
+    let wrapper: ({ children }: { children: JSX.Element }) => JSX.Element, mock_DBot_store: RootStore | undefined;
+    beforeAll(() => {
+        const mock_store = mockStore({
+            ui: {
+                notification_messages_ui: jest
+                    .fn()
+                    .mockReturnValue(<p>Mocked Notification Message</p>) as React.ElementType,
+            },
+        });
+        mock_DBot_store = mockDBotStore(mock_store, mock_ws);
+
+        wrapper = ({ children }: { children: JSX.Element }) => (
+            <StoreProvider store={mock_store}>
+                <DBotStoreProvider ws={mock_ws} mock={mock_DBot_store}>
+                    {children}
+                </DBotStoreProvider>
+            </StoreProvider>
+        );
+    });
+
+    it('should render BotNotificationMessages', () => {
+        const { container } = render(<BotNotificationMessages />, {
+            wrapper,
+        });
+        expect(container).toBeInTheDocument();
+        expect(screen.getByTestId('dt_notifications_container')).toHaveClass('notifications-container');
+    });
+
+    it('should apply notifications-container__dashboard class when active tab is 0 and is_info_panel_visible is true', () => {
+        mock_DBot_store!.dashboard.is_info_panel_visible = true;
```

Comment:
@vinu-deriv Bhai instead of directly mutating the state here can we use the setInfoPanelVisibility() function here? I mean by directly mutating an observable value in the MobX store we actually bypass MobX's built-in mechanisms for tracking changes and notifying observers right?